src/main/java/banana/Parser.java

```diff
@@ -0,0 +1,281 @@
+
+    /**
+     * Finds the task that contains
+     * or matches the user's input.
+     *
+     * @param tasks the list of tasks.
+     */
+    public void findCommand(TaskList tasks) {
+        TaskList newTasks = new TaskList(new ArrayList<>());
+        if (input.contains("find")) {
+            String item = input.split(" ")[1];
+            for (int i = 0; i < tasks.size(); i++) {
+                if (tasks.getTask(i).getDescription().contains(item)) {
+                    newTasks.addTask(tasks.getTask(i));
+                }
+            }
+            displayLabel(
+                    "Here are the matching tasks in your list: \n" +
+                            "     " + getItems(newTasks));
+        }
+    }
+
+    /**
+     * Gets the list of tasks.
+     *
+     * @param tasks the list of tasks.
+     * @return the tasks in string format.
+     */
+    public String getItems(TaskList tasks) {
+        String collection = "";
+        for (int index = 0; index < tasks.size(); index++) {
+            if (index != 0) {
+                collection += "     ";
+            }
+            String info = tasks.getTask(index).toString();
+            collection += Integer.toString(index + 1) + "." + info;
+            if (index != tasks.size() - 1) {
+                collection += "\n";
+            }
+        }
+        return collection;
+    }
+
+    /**
+     * Gets the date and time in correct
+     * notation/format if necessary.
+     *
+     * @param info the task information.
+     * @param input the user input.
+     * @param type the Task type.
+     * @return the new Task.
+     */
+    public Task getDateAndTime(String[] info, String input, String type) {
+        String[] potentialDate = info[1].split(" ");
+        LocalDate date = null;
+        if (!getTime(potentialDate[0]).equals("")) {
+            info[1] = getTime(potentialDate[0]);
+        } else if (potentialDate.length > 1 &&
+                !getTime(potentialDate[1]).equals("")) {
+            potentialDate[1] = getTime(potentialDate[1]);
+            info[1] = potentialDate[0] + " " + potentialDate[1];
+        }
+        if (potentialDate.length > 1 && potentialDate[0].contains("/")) {
+            date = LocalDate.parse(parseDates(potentialDate[0]));
+        }
+        if (type.equals("deadline")) {
+            if (date != null) {
+                return new Deadline(info[0], date, potentialDate[1]);
+            } else {
+                return new Deadline(info[0], info[1]);
+            }
+        } else {
+            if (date != null) {
+                return new Event(info[0], date, potentialDate[1]);
+            } else {
+                return new Event(info[0], info[1]);
+            }
+        }
+    }
+
+    /**
+     * Changes the date format to be
+     * parsed by LocalDate.
+     *
+     * @param date the old-format date.
+     * @return the date.
+     */
+    public String parseDates(String date) {
+        String[] sep = date.split("/");
```

Comment:
As readers may not be able to infer the purpose of this variable from the abbreviated variable name, perhaps it would be better if a lengthier name was used to describe this variable instead? I noticed this same issue in some other places as well.